nupic/research/frameworks/continuous_learning/multihead/subdataset.py

```
@@ -0,0 +1,191 @@
+    continuous learning setup
+    :param dataset_name: name of the dataset to retrieve
+    :type dataset_name: str
+    :param scenario: continuous learning setup, one of {'task', 'domain', 'class'}
+    :type scenario: str
+    :param download: whether or not to download the dataset
+    :type download: bool
+    :return: tuple of (list of pytorch dataset, list of pytorch dataset)
+    """
+    # specify image-transformations to be applied
+    dataset_transform = transforms.Compose([
+        transforms.ToTensor(),
+    ])
+
+    if dataset_name == "splitGSC":
+        dataset_class = GSC
+    elif dataset_name == "splitMNIST":
+        dataset_class = MNIST
+
+    full_train_set = dataset_class("~/nta/continual-learning/datasets/mnist/",
+                                   train=True, transform=dataset_transform,
+                                   target_transform=None, download=download)
+    full_test_set = dataset_class("~/nta/continual-learning/datasets/mnist/",
+                                  train=False, transform=dataset_transform,
+                                  target_transform=None, download=download)
+
+    train_datasets, test_datasets = [], []
+
+    for labels in [[0, 1], [2, 3], [4, 5], [6, 7], [8, 9]]:
+
+        if scenario == "domain":
+            target_transform = transforms.Lambda(lambda y, x=labels[0]: y - x)
+        else:
+            target_transform = None
+
+        train_datasets.append(SubDataset(full_train_set, labels,
+                                         target_transform=target_transform))
+        test_datasets.append(SubDataset(full_test_set, labels,
+                                        target_transform=target_transform))
+
+    return train_datasets, test_datasets
+
+
+class GSC(Dataset):
+    """ Google Speech Commands dataset. """
+
+    def __init__(self, root=".", train=True, transform=None, target_transform=None,
+                 download=False):
```

This download flag is not being used. If it is here to maintain consistency with the torchvision API, it should throw an exception if the dataset is not available, warning the user that the download flag is being ignored and the dataset has to be manually downloaded.

---

Noted.